taglib/tagunion.cpp

```diff
@@ -119,6 +119,40 @@ namespace TagLib
   }
 
   template <size_t COUNT>
+  PropertyMap TagUnion<COUNT>::setProperties(const PropertyMap &properties)
+  {
+    //Record unassigned properties for each tag in the union
+    std::vector<PropertyMap> returnCandidates;
+    for (size_t i = 0; i < COUNT; ++i) {
+      if (d->tags[i])
+        returnCandidates.insert(returnCandidates.end(), d->tags[i]->setProperties(properties));
+    }
+
+    if (!returnCandidates.empty()) {
+      //Only one tag present, return its unassigned properties
+      if (returnCandidates.size() == 1) {
+        return returnCandidates.front();
+      }
+
+      //Multiple tags in union: if a property has been assigned in any member tag, remove it from ignored properties to return
```

Comment:
I haven't looked at this in detail at all, but just from a quick glance the super long lines need to be truncated.  TagLib generally tries to keep things to 80-ish columns.